File: appinventor/components/src/com/google/appinventor/components/common/HorizontalAlignment.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2020 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.common;

import java.util.HashMap;
import java.util.Map;

/**
 * Defines a HorizontalAlignment used by a variety of classes including Form, VerticalArrangement,
 * and Marker.
 */
public enum HorizontalAlignment implements OptionList<Integer> {
  Left(1),
  Center(3), // Yes this is correct.
  Right(2);

  private final int value;

  HorizontalAlignment(int value) {
    this.value = value;
  }

  public Integer toUnderlyingValue() {
    return value;
  }

  private static final Map<Integer, HorizontalAlignment> lookup = new HashMap<>();

  static {
    for (HorizontalAlignment alignment : HorizontalAlignment.values()) {
      lookup.put(alignment.toUnderlyingValue(), alignment);
    }
  }

  public static HorizontalAlignment fromUnderlyingValue(Integer alignment) {
    return lookup.get(alignment);
  }
}

File: appinventor/components-ios/src/HorizontalAlignment.swift
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright © 2025 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// https://www.apache.org/licenses/LICENSE-2.0

@objc public class HorizontalAlignment: NSObject, OptionList {
  @objc public static let Left = HorizontalAlignment(1)
  @objc public static let Center = HorizontalAlignment(3)
  @objc public static let Right = HorizontalAlignment(2)

  private static let LOOKUP: [Int32: HorizontalAlignment] = generateOptionsLookup(Left, Center, Right)

  let value: Int32

  @objc private init(_ value: Int32) {
    self.value = value
  }

  @objc public func toUnderlyingValue() -> AnyObject {
    return value as AnyObject
  }

  @objc public static func fromUnderlyingValue(_ value: Int32) -> HorizontalAlignment? {
    return LOOKUP[value]
  }
}
